### PR TITLE
Element Call permission tweaks

### PR DESCRIPTION
**Pull request by stefanceriu**
vector-im/element-meta/issues/2230 - Specify `call.member` default po…wer levels when creating a room
vector-im/element-meta/issues/2229 - Take into account permissions when configuring the room screen call button

### CODE REVIEW

File: ElementX/Sources/Services/Client/ClientProxy.swift [hunk 1]
```diff
@@ -575,6 +584,22 @@ class ClientProxy: ClientProxyProtocol {
...
+              stateDefault: nil,
+              ban: nil,
+              kick: nil,
+              redact: nil,
+              invite: nil,
+              notifications: nil,
+              users: [:],
+              events: [
+                  "m.call.member": Int32(0),
+                  "org.matrix.msc3401.call.member": Int32(0)
```
Comment (toger5) [inline ElementX/Sources/Services/Client/ClientProxy.swift:600]: This is really good, that we have both here (the unstable prefix + the final one)

**Review by toger5 — APPROVED** [review]
Looks reasonable. It does make sense and is easy to follow even though I am not familiar with the codebase.

Have you tested this (There are no rust sdk unit tests for this part since we dont have them for the FFI and most of this is FFI code)

Just one sanity test should be enough (see if a new room is having the correct entry in power_levels, there are not many cases but this would guarantee that there is no insane typo somewhere...)

File: ElementX/Sources/Services/Room/RoomProxy.swift [hunk 2]
```diff
@@ -396,6 +396,15 @@ class RoomProxy: RoomProxyProtocol {
         }
     }
     
+    func canUserJoinCall(userID: String) async -> Result<Bool, RoomProxyError> {
+        do {
+            return try await .success(room.canUserSendState(userId: userID, stateEvent: .roomMemberEvent))
```
Comment (pixlwave) [inline ElementX/Sources/Services/Room/RoomProxy.swift:401]: Can we add a comment about why I need to be able to send a room member event in order to join a call? It feels very counter-intuitive just looking at it.

File: ElementX/Sources/Services/Client/ClientProxy.swift [hunk 3]
```diff
@@ -575,6 +584,22 @@ class ClientProxy: ClientProxyProtocol {
             return (nil, nil)
         }
     }
+    
+    private static var defaultRoomCreationPowerLevels: PowerLevels {
```
Comment (pixlwave) [inline ElementX/Sources/Services/Client/ClientProxy.swift:588]: I kind of expected to find this static var at the top of the file. Not a biggy, just unexpected.

Comment (pixlwave) [reply ElementX/Sources/Services/Client/ClientProxy.swift:588]: Also, a second thought for a follow-up PR: could this exist in the SDK? So we could simply pass `powerLevelContentOverride: .callOverride` or similar to the creation parameters?

**Review by pixlwave — APPROVED** [review]
Nice 👌

File: ElementX/Sources/Services/Room/RoomProxy.swift (hunk 2 above)
Comment (toger5) [reply ElementX/Sources/Services/Room/RoomProxy.swift]: Sorry for missing this in my review...

**Pull request merged**